private/signer/v4/v4.go

```diff
@@ -326,7 +326,7 @@ func (v4 *signer) buildCanonicalHeaders(r rule, header http.Header) {
 			headerValues[i] = "host:" + v4.Request.URL.Host
 		} else {
 			headerValues[i] = k + ":" +
-				strings.Join(v4.Request.Header[http.CanonicalHeaderKey(k)], ",")
+				strings.Join(v4.signedHeaderVals[k], ",")
```

Comment:
It seems that the headers that do not satisfy the rules (and therefore, are not passed through) would also not get added `headers` slice which is used to iterate through on the second loop.

It seems that it is guaranteed that every `k` within the second loop would be in `strings.Join(v4.signedHeaderVals[k], ",")`.